Implement FixIt for C#

mispencer opened the pull request:
Implement https://github.com/Valloric/ycmd/pull/167 for C#.

I have signed the CLA.

Valloric commented on ycmd/tests/testdata/testy/FixItTestCase.cs:
```diff
@@ -0,0 +1,13 @@
+			var str = "";
+			str.EndsWith("A");
+			var i = 5;
+			var j = i + 5;
```
Consider adding a comment in the file describing the expected fix-it.

Valloric commented:
This looks nice, thanks for sending it in!

@puremourning I'm interested in your thoughts on this implementation since you implemented the libclang support.

mispencer commented:
I updated the pull request to resolve issues if there wasn't any text changes or the text change affected the first or last character in the buffer. I also added some tests.

Regarding comments in the test data file: I would rather not, since comments are not negligible here - they affect the fix output.

Valloric commented:
> Regarding comments in the test data file: I would rather not, since comments are not negligible here - they affect the fix output.

Oh, that makes sense.

puremourning commented:
@Valloric this looks ok to me. I've tested with the client changes and incorporated them into PR https://github.com/Valloric/YouCompleteMe/pull/1555

Valloric commented:
Thanks you both!

@homu r+

homu commented:
:pushpin: Commit 88d2e42 has been approved by `Valloric`

<!-- @homu r=Valloric 88d2e42e2c2d69769eeb119279ce172998655359 -->

homu commented:
:hourglass: Testing commit 88d2e42 with merge 1855101...

homu commented:
:sunny: Test successful - [status](https://travis-ci.org/Valloric/ycmd/builds/72049684)
